### PR TITLE
Add transliterations

### LINKED ISSUE #483: Create transliteration module or use existing one.
To be able to transliterate special characters in URL's as requested in #293, we need a module that can transliterate special characters in strings.

### Our requirements
- We need a module that takes a string and a locale and returns the transliterated version of the string, based on that locale.
- We want the module to strive for full coverage of every available locale. Of course it's fine if the module doesn't cover everything yet, but modules that were created only to fix it for i.e. cyrillic are not interesting to us (except to compare our own data with).
- We don't want defaults or reuse of characters in any way, shape or form. Transliteration tables should be specific for every language.

### Reuse or build our own?

It seems there are already many transliteration modules. Still, they all seem to be reinventing the wheel and many are not transparent about where they base their transliteration tables on. All of them seem to do it for a couple of languages, but not for all. Some have a lot of languages, but are not localized as well or are incomplete for some languages. 

If we can find a module that fits our requirements, then we should use that. If we can find a module that fits nearly all the requirements, we should consider if we can contribute to it in a way that gets us very far very fast. 

There are quite some advantages to owning the module ourselves:

- We can develop it more rapidly.
- We can control the specification.
- We can control the way it is packaged and modularized.

Since transliteration tables are not likely subject to a lot of change, the risk of high maintenance is relatively low.

### PR DESCRIPTION
**Pull request by IreneStr**
Add transliterations for:

- Spanish
- Polish
- German
- Nynorsk
- Bokmål
- Swedish
- Finnish
- Danish
- Turkish
- Latvian
- Icelandic
- Faroese
- Czech
- Russian
- Esperanto
- Afrikaans
- Catalan
- Asturian
- Aragonese
- Aymara
- English
- French
- Italian
- Dutch
- Bambara
- Ukrainian

Fixes #483 

For testing check the unit tests.

### CODE REVIEW

File: js/config/transliterations.js [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+ * The function returning an array containing transliteration objects, based on the given locale.
+ *
+ * @param {string} locale The locale.
+ * @returns {Array} An array containing transliteration objects.
+ */
+module.exports = function( locale ) {
+	if ( isUndefined( locale ) ) {
+		return [];
+	}
+		switch( getLanguage( locale ) ) {
```
Comment (andizer) [inline js/config/transliterations.js]: Indentation it too much.

**Comment by andizer** [comment]
CR done. One big feedback thing:
I would choose for a big array with all the transliterations. This array should be outside the module.exports thing. 

```js
var transliterations = { en : { }  }; 

module.exports = function() {
    if ( isUndefined( locale ) ) {
        return [];
    }

    var language = getLanguage( locale );
 
    if ( check here if the language exists in the transliterations object ) {
        return transliterations[ language ]; 
    }

    return [];
```

**Comment by andizer** [comment]
CR done 👍

**Comment by rensw90** [comment]
Acceptance done 💯

**Pull request merged**